### PR TITLE
ci(github): add sync-generated-files workflow

### PR DESCRIPTION
**Pull request by joshblack**
Updates our CI config so that we use GitHub Actions to check-in generated artifacts instead of CircleCI

#### Changelog

**New**

- Add new GitHub workflow for syncing generated files

**Changed**

- CircleCI no longer has an artifacts workflow

**Removed**

### CODE REVIEW

File: .github/workflows/sync-generated-files.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  release:
+    runs-on: macOS-latest
+    steps:
+      - uses: actions/checkout@master
+      - name: Use Node.js 10.x
+        uses: actions/setup-node@v1
+        with:
+          version: 10.x
+      - name: Install dependencies
+        run: yarn install --offline
```
Comment (jnm2377) [inline .github/workflows/sync-generated-files.yml:16]: Looks good. I was just curious why in the circle ci config we had to do `--frozen-lockfile` but not here?

File: .circleci/config.yml [hunk 2]
```diff
@@ -25,58 +25,8 @@ jobs:
...
-  artifacts:
-    docker:
-      - image: circleci/node:10.15-browsers
-    working_directory: ~/repo
-    steps:
-      - checkout
-      - run:
-          name: Install yarn
-          command: |
-            curl -o- -L https://yarnpkg.com/install.sh | bash -s -- --version 1.16.0
```
Comment (jnm2377) [inline .circleci/config.yml:37]: also curious why we don't have to install yarn in github actions?

**Review by jnm2377 — APPROVED** [review]

File: .github/workflows/sync-generated-files.yml (hunk 1 above)
Comment (joshblack) [reply .github/workflows/sync-generated-files.yml:16]: They're essentially the same thing here, `--offline` means we only download from a known folder where-as `--frozen-lockfile` will fail if `package.json` specifies a version outside of what is in `yarn.lock`. Really we only need the former haha

File: .circleci/config.yml (hunk 2 above)
Comment (joshblack) [reply .circleci/config.yml:37]: We check in our Yarn versions (under `.yarn/releases`) and so as long as the docker image has yarn it should redirect to the latest version 😄

File: .github/workflows/sync-generated-files.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      - master
+jobs:
+  release:
+    runs-on: macOS-latest
+    steps:
+      - uses: actions/checkout@master
+      - name: Use Node.js 10.x
+        uses: actions/setup-node@v1
+        with:
+          version: 10.x
```
Comment (vpicone) [inline .github/workflows/sync-generated-files.yml]: I _believe_ this needs to be `node-version` now https://github.com/actions/setup-node#usage

**Review by vpicone — CHANGES_REQUESTED** [review]

File: .github/workflows/sync-generated-files.yml (hunk 3 above)
Comment (joshblack) [inline .github/workflows/sync-generated-files.yml]:
```suggestion
          node-version: '10.x'
```

Comment (joshblack) [reply .github/workflows/sync-generated-files.yml]: Should be good now!

**Review by vpicone — APPROVED** [review]

**Pull request merged**